### PR TITLE
Adding tests for core utilities Solver class

### LINKED ISSUE #62: Add unit tests for Solver class in core utilities
Solver needs unit tests

### PR DESCRIPTION
**Pull request by cainja**
The public static methods in core utilities needed tests

closes #62.

### CODE REVIEW

File: test/arcade/core/util/SolverTest.java [hunk 1]
```diff
@@ -0,0 +1,226 @@
+package arcade.core.util;
+
+import org.junit.jupiter.api.Test;
+import arcade.core.util.Solver.Equations;
+import arcade.core.util.Solver.Function;
+import static org.junit.jupiter.api.Assertions.*;
+
+public class SolverTest {
+    @Test
+    public void testConstructor_called_returnsException() {
```
Comment (jessicasyu) [inline test/arcade/core/util/SolverTest.java:10]: Can simply to:

```java
@Test
public void testConstructor_called_returnsException() {
    assertThrows(UnsupportedOperationException.class, Solver::new);
}
```

File: test/arcade/core/util/SolverTest.java [hunk 2]
```diff
@@ -0,0 +1,226 @@
...
+        assertThrows(
+                UnsupportedOperationException.class,
+                () -> {
+                    new Solver();
+                });
+    }
+
+    @Test
+    public void testEuler_calledWithEquations_returnsAnswer() {
```
Comment (jessicasyu) [inline test/arcade/core/util/SolverTest.java:19]: Can simplify all the `Equations` constructors to:

```java
Equations e = (t, y) -> {
    ...
};
```

File: test/arcade/core/util/SolverTest.java [hunk 3]
```diff
@@ -0,0 +1,226 @@
...
+                    }
+                };
+
+        double[] y = Solver.euler(e, 0, new double[] {1, 0}, 1, 0.01);
+        assertEquals(Math.exp(-1), y[0], 0.01);
+        assertEquals(1 - Math.exp(-1), y[1], 0.01);
+    }
+
+    @Test
+    public void testRungeKutta_calledWithEquations_returnsAnswer() {
```
Comment (jessicasyu) [inline test/arcade/core/util/SolverTest.java:53]:
```suggestion
    public void testRungeKutta_calledWithSimpleEquations_returnsAnswer() {
```

For consistency with the "complex" tests?

**Review by jessicasyu — APPROVED** [review]
Looks good! There are a couple missed branches in the coverage, if you want to revisit those cases

File: test/arcade/core/util/SolverTest.java (hunk 2 above)
Comment (cainja) [reply test/arcade/core/util/SolverTest.java:19]: maybe I'm misunderstanding but the interface for equations requires a dydt method

**Comment by cainja** [comment]
Improved coverage but am not sure about a good way to test the adaptive step in the Cash Karp method. If you thumbs up this comment I'll just merge.

**Review by jessicasyu — APPROVED** [review]

**Pull request merged**